cpu/stm32_common/periph/dma.c

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (C) 2017 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_cortexm_common
+ * @{
+ *
+ * @file
+ * @brief       Low-level DMA driver implementation
+ *
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+
+#include "periph_cpu.h"
+#include "periph_conf.h"
+#include "mutex.h"
+#include "assert.h"
+
+#if defined(DMA_NUMOF) && (defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4) || defined(CPU_FAM_STM32F7))
+
+#define DMA_STREAM_IT_MASK   (DMA_LISR_FEIF0 | DMA_LISR_DMEIF0 | \
+                              DMA_LISR_TEIF0 | DMA_LISR_HTIF0 | \
+                              DMA_LISR_TCIF0)
+
+struct dma_ctx {
+    mutex_t conf_lock;
+    mutex_t sync_lock;
+    uint16_t len;
+};
+
+static struct dma_ctx dma_ctx[DMA_NUMOF];
+
+static inline uint32_t dma_all_flags(dma_t dma)
+{
+    assert(dma < DMA_NUMOF);
+
+    switch (dma_config[dma].stream & 0x3) {
+    case 0: /* 0 and 4 */
```

Comment:
the case block shoud be indented a well (even I don't really like it, that's part of RIOT code conventions...)

Reply:
I don't see were in the coding convention we should indent the case. Is there a consensus on that?

Reply:
Well looking at other places in the code, the indented case is the most common.